cloudflare-lb/main.tf

```diff
@@ -0,0 +1,155 @@
+# Fetch the record if it exists
+data "cloudflare_zones" "subspace" {
+  filter {
+    name   = var.domain
+    status = "active"
+  }
+}
+
+# rpc node
+data "cloudflare_record" "existing_rpc_record" {
+  zone_id  = data.cloudflare_zones.subspace.id
+  hostname = "rpc.${var.network}.${var.domain}"
+  type     = "A"
+}
+
+resource "cloudflare_record" "rpc-0" {
+  count   = length(data.cloudflare_record.existing_rpc_record.hostname) > 0 ? 0 : 1
+  zone_id = var.zone_id
+  name    = "rpc"
+  value   = "52.91.27.239"
```

Comment:
are there the load balancer IPs cloudfalre provided or the actual rpc IPs ?
If it is the latter, then shouldn't they come from the terrform config instead ?